Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/presencetracing/organizer/warn/duration/TraceLocationWarnDurationFragment.kt

```diff
@@ -0,0 +1,168 @@
+import com.google.android.material.datepicker.DateValidatorPointBackward
+import com.google.android.material.datepicker.MaterialDatePicker
+import com.google.android.material.timepicker.MaterialTimePicker
+import com.google.android.material.timepicker.TimeFormat
+import com.google.android.material.transition.MaterialSharedAxis
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.contactdiary.util.hideKeyboard
+import de.rki.coronawarnapp.databinding.TraceLocationOrganizerWarnDurationFragmentBinding
+import de.rki.coronawarnapp.ui.durationpicker.DurationPicker
+import de.rki.coronawarnapp.ui.durationpicker.toContactDiaryFormat
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import org.joda.time.LocalDate
+import org.joda.time.LocalTime
+import javax.inject.Inject
+
+class TraceLocationWarnDurationFragment :
+    Fragment(R.layout.trace_location_organizer_warn_duration_fragment),
+    AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val binding: TraceLocationOrganizerWarnDurationFragmentBinding by viewBinding()
+    private val navArgs by navArgs<TraceLocationWarnDurationFragmentArgs>()
+    private val viewModel: TraceLocationWarnDurationViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as TraceLocationWarnDurationViewModel.Factory
+            factory.create(navArgs.traceLocation)
+        }
+    )
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        enterTransition = MaterialSharedAxis(MaterialSharedAxis.Z, true)
+        returnTransition = MaterialSharedAxis(MaterialSharedAxis.Z, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.state.observe2(this) { uiState ->
+            with(binding) {
+                if (uiState.description != null) {
+                    description.text = uiState.description
+                }
+
+                if (uiState.startDateTime != null && uiState.endDateTime != null) {
+
+                    val startTime = uiState.startDateTime!!.toDateTime()
+                    val endTime = uiState.endDateTime!!.toDateTime()
+
+                    eventDate.isGone = false
+
+                    val startDay = startTime.toLocalDate().toString("dd.MM.yyyy")
+                    val startHour = startTime.toLocalTime().toString("HH:mm")
+                    val endDay = endTime.toLocalDate().toString("dd.MM.yyyy")
+                    val endHour = endTime.toLocalTime().toString("HH:mm")
```

Comment:
I think we have some extension functions for formatting. `toShortDayFormat` and `toShortTimeFormat`